code/game/machinery/computer/singulo_monitor.dm

```diff
@@ -0,0 +1,205 @@
+		data["singulo_potential_stage"] = (active.allowed_size + 1) / 2
+		data["singulo_energy"] = active.energy
+
+		switch(active.allowed_size)
+			if(STAGE_ONE)
+				data["singulo_high"] = STAGE_TWO_THRESHOLD
+				data["singulo_low"] = 0
+			if(STAGE_TWO)
+				data["singulo_high"] = STAGE_THREE_THRESHOLD
+				data["singulo_low"] = STAGE_TWO_THRESHOLD
+			if(STAGE_THREE)
+				data["singulo_high"] = STAGE_FOUR_THRESHOLD
+				data["singulo_low"] = STAGE_THREE_THRESHOLD
+			if(STAGE_FOUR)
+				data["singulo_high"] = STAGE_FIVE_THRESHOLD
+				data["singulo_low"] = STAGE_FOUR_THRESHOLD
+			else
+				data["singulo_high"] = STAGE_SIX_THRESHOLD
+				data["singulo_low"] = STAGE_FIVE_THRESHOLD
+
+		var/list/generators = list()
+		var/index = 1
+		for(var/obj/machinery/field/generator/generator in field_gens)
+			generators.Add(list(list(
+				"charge" = generator.power,
+				"gen_index" = index++
+			)))
+		data["generators"] = generators
+
+	else
+		var/list/singulos = list()
+		for(var/I in singularities)
+			var/obj/singularity/S = I
+			var/area/A = get_area(S)
+			if(!A)
+				continue
+
+			singulos.Add(list(list(
+				"area_name" = A.name,
+				"energy" = S.energy,
+				"stage" = (S.current_size + 1) / 2,
+				"singularity_id" = S.singulo_id
+			)))
+
+		data["active"] = FALSE
+		data["singularities"] = singulos
+
+	return data
+
+/**
+  * Supermatter List Refresher
+  *
+  * This proc loops through the list of supermatters in the atmos SS and adds them to this console's cache list
+  */
+/obj/machinery/computer/singulo_monitor/proc/refresh()
+	singularities = list()
+	var/turf/T = get_turf(ui_host()) // Get the UI host incase this ever turned into a singularity monitoring module for AIs to use or something
+	if(!T)
+		return
+	for(var/obj/singularity/S in GLOB.singularities)
+		// not within coverage, not on a tile.
+		if(!(is_station_level(S.z) || is_mining_level(S.z) || atoms_share_level(S, T) || !issimulatedturf(S.loc) || istype(active)))
+			continue
+		singularities.Add(S)
+
+	if(!(active in singularities))
+		active = null
+
+/obj/machinery/computer/singulo_monitor/proc/send_alerts()
+	// Breach alerts
+	if(active.current_size > (last_size + 2) || active.current_size >= STAGE_FIVE)// We should only see a singulo grow 2 stages at once when breaching containment.
+		singu_radio.autosay("<b>Warning: The singularity in [get_area(active)] has exceeded containment field limits!</b>", name, breach_channel)
+	for(var/obj/machinery/field/generator/gen in field_gens)
+		if(!gen || (gen.active < 2))
+			singu_radio.autosay("<b>Warning: The containment field of the singularity in [get_area(active)] has been disabled!</b>", name, breach_channel)
+			return
+
+	if(active.energy > last_energy)// We only want to give warnings while the situation is getting worse.
+		if(active.energy >= (STAGE_FIVE_THRESHOLD - 100))
+			singu_radio.autosay("<b>Warning: The singularity in [get_area(active)] approaching an uncontainable level!</b>", name, warning_channel)
```

Comment:
This is likely to just be `Space` most of the time, isn't it? I don't know if you need an area for this announcement. If you need to be more specific, you screwed up somewhere along the line.

Reply:
removed it